library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -32,8 +32,10 @@ impl Command {
 
         let (ours, theirs) = self.setup_io(default, needs_stdin)?;
 
-        if let Some(ret) = self.posix_spawn(&theirs, envp.as_ref())? {
-            return Ok((ret, ours));
+        if self.get_groups().is_none() {
```

Comment:
Existing checks like that are in inside the posix_spawn method, so I would move this one there as well:
https://github.com/rust-lang/rust/blob/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/std/src/sys/unix/process/process_unix.rs#L286-L292